### PR TITLE
[sp-184] search by location

### PR DESCRIPTION
**Pull request by shawn-cx-li**
https://aussiecommerce.atlassian.net/browse/SP-184

### CODE REVIEW

File: src/lib/redisFunctions.ts [hunk 1]
```diff
@@ -297,3 +295,24 @@ export async function getAllOffersWithPackagesFromKeys(
...
+  destinationKey: string,
+  region: string
+): Promise<string[]> {
+  const offers: any[] = await getAvailableOffersByDestinationKeys(
+    [destinationKey],
+    region
+  );
+
+  const offerIds = offers
+    .map((offer: any) => [
```
Comment (joshuapaling) [inline src/lib/redisFunctions.ts:309]: I don't think we should add types if they're just `any`. Better to not have types (and have the linter tell us where we're missing with a warning) than having meaningless types.

Comment (shawn-cx-li) [reply src/lib/redisFunctions.ts]: sounds good, removed `any`

File: src/controllers/offerSearchController.ts [hunk 2]
```diff
@@ -43,9 +43,8 @@ export async function searchOffersByLocationCapacityAndTravelDate(
   req: Request,
   res: Response
 ): Promise<Response | void> {
-  const { location, adults, children, infants, checkIn, checkOut } = req.query;
-
-  const locationTerm = location ? location.toString() : "";
+  const placeId: any = req.params.placeId;
```
Comment (joshuapaling) [inline src/controllers/offerSearchController.ts:46]: In this case we know it's a string - so do `placeId: string`

File: src/lib/redisFunctions.ts [hunk 3]
```diff
@@ -297,3 +295,24 @@ export async function getAllOffersWithPackagesFromKeys(
...
+  );
+
+  const offerIds = offers
+    .map(offer => [
+      ...offer.available_offers.continent,
+      ...offer.available_offers.country,
+      ...offer.available_offers.administrative_area_level_1,
+      ...offer.available_offers.local
+    ])
+    .reduce((prev: any, curr: any) => [...prev, ...curr], []);
```
Comment (joshuapaling) [inline src/lib/redisFunctions.ts:315]: remove any here too

**Pull request merged**